org.idempiere.test/src/org/idempiere/test/model/MSequenceTest.java

```diff
@@ -0,0 +1,137 @@
+package org.idempiere.test.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.sql.Timestamp;
+import java.util.Properties;
+
+import org.compiere.model.MBPartner;
+import org.compiere.model.MDocType;
+import org.compiere.model.MOrder;
+import org.compiere.model.MSequence;
+import org.compiere.process.DocAction;
+import org.compiere.util.Env;
+import org.idempiere.test.AbstractTestCase;
+import org.idempiere.test.DictionaryIDs;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+/**
+ * Tests for {@link org.compiere.model.MSequence} class.
+ * Specifically testing document number generation using AD_Sequence
+ * with new context variable keys feature (/K flag).
+ */
+public class MSequenceTest extends AbstractTestCase {
+
+    private static final String TEST_PREFIX = "TestSeq-";
+    private static final String TEST_SUFFIX = "-End";
+    private static final String TEST_DATE = "2025-06-13 10:00:00";
+    private static final String TEST_DATE_yyyyMM = "202506";
+
+    @Test
+    public void testDocumentNoWithContextVariableKey() {
+        Properties ctx = Env.getCtx();
+        String trxName = getTrxName();
+
+        // Create test sequence
+        MSequence seq = new MSequence(ctx, 0, trxName);
+        seq.setName("Test Sequence with Context Key");
+        seq.setDescription("Test sequence for document numbers with context variable key");
+        seq.setPrefix(TEST_PREFIX + "@Updated<yyyyMM>/K@"); // /K flag indicates context variable use
```

Comment:
This will not always match with `TEST_DATE_yyyyMM = "202506"` as Updated is fill with current system date instead of from "#Date". In my local testing, I have to change it to `@DateOrdered<yyyyMM>/K@` to pass the unit test. `@Updated<yyyyMM>/K@` with evaluate to "202507" (current system date) instead .